manifests/install.pp

```diff
@@ -2,10 +2,30 @@
 class nats::install {
   contain "nats::install::${nats::service_type}"
 
+  $_manage_group = $nats::manage_group and $nats::group != "root"
+  if $_manage_group {
+    group { $nats::group:
+      ensure    => present,
+      allowdupe => false,
+    }
+  }
+  if $nats::manage_user and $nats::user != "root" {
+    $_user_require = $_manage_group ? {
+      true    => Group[$nats::group],
+      default => undef
+    }
+    user { $nats::user:
+      ensure    => present,
+      gid       => $nats::group,
+      allowdupe => false,
+      require   => $_user_require
+    }
```

Comment:
This require is not needed. Puppet auto requires this so this and the temp cars and selector can all go